app/src/main/java/dev/animeshvarma/sigil/MainActivity.kt

```
@@ -34,17 +65,43 @@ class MainActivity : AppCompatActivity() {
     private lateinit var prefs: SigilPreferences
 
     private val isContentHidden = mutableStateOf(true)
+    private val isAppInForeground = mutableStateOf(true)
+    private val isWindowFocused = mutableStateOf(true)
+
+    private var screenCaptureCallback: Any? = null
+
+    private lateinit var secureOverlayView: FrameLayout
```

detekt (1.23.8)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)